### PR TITLE
Pie chart component

### PR DESCRIPTION
**Pull request by mpisacic**

### CODE REVIEW

File: src/components/PieChart/PieChart.tsx [hunk 1]
```diff
@@ -0,0 +1,195 @@
...
+            .text(d.data.text);
+    }
+
+    private calculateClass(d: DataType) {
+        if (d.label === 'free') { return 'disk-usage-free'; }
+        
+        let className;
+
+        if (this.props.statusInterval) {
+            const okInterval = this.props.statusInterval.filter((d) => d.status === 'ok')[0];
```
Comment (vinkobedek) [inline src/components/PieChart/PieChart.tsx]: this entire region of code is awfully specific. Shouldn't this component be a bit more generic?

Comment (mpisacic) [inline src/components/PieChart/PieChart.tsx]: We could make component drawing some random (or specified) colors that aren't tied up to classes or anything, so every part of chart will be in some of specified color.

Comment (vinkobedek) [inline src/components/PieChart/PieChart.tsx]: usually there is a default pallete defined, but you can override it for your "series".

File: src/components/PieChart/PieChart.tsx [hunk 2]
```diff
@@ -0,0 +1,179 @@
+import * as React from 'react';
+import * as d3 from 'd3';
+import * as classNames from 'classnames';
+import { Label } from '../Label/Label';
+import { IPieChartProps } from './PieChart.props';
+
+export type DataType = {
```
Comment (hbagarich) [inline src/components/PieChart/PieChart.tsx]: can this be replaced with IPieChartData interface?

File: src/components/PieChart/PieChart.tsx (hunk 1 above)
Comment (mpisacic) [inline src/components/PieChart/PieChart.tsx]: Added default pallette and classes so color can be redefined in container component above this one.

File: src/components/PieChart/PieChart.tsx (hunk 2 above)
Comment (mpisacic) [inline src/components/PieChart/PieChart.tsx]: Did you mean replacing DataType with IPieChartData interface?

File: src/components/PieChart/PieChart.tsx [hunk 3]
```diff
@@ -170,10 +170,21 @@ export class PieChart extends React.Component<IPieChartProps, any> {
...
-        return d3.pie<DataType>().sort(null).value((d): any => d.value);
+        return d3.pie<IPieChartData>().sort(null).value((d): any => d.value);
     }
 
     private createColorPallette() {
         return d3.scaleOrdinal(d3.schemeCategory10);
     }
+
+    private _checkNameLen(name: string) : string {
+        if (name.length > 5) {
```
Comment (hbagarich) [inline src/components/PieChart/PieChart.tsx]: can we use text-overflow: ellipsis for these cases?

Comment (mpisacic) [inline src/components/PieChart/PieChart.tsx]: I'm not sure but probably, I'll try it

**Pull request merged**